include/circt/Dialect/FIRRTL/FIRRTLCanonicalization.td

```diff
@@ -349,6 +355,11 @@ def NarrowMuxRHS : Pat <
   (MoveNameHint $old, (CatPrimOp (MuxPrimOp $cond, $a, $c), $b)),
   [(EqualTypes $lhs, $rhs), (KnownWidth $lhs)]>;
 
+def CatDoubleConst : Pat <
+  (CatPrimOp:$old $cst1, (CatPrimOp $cst2, $v)),
+  (MoveNameHint $old, (CatPrimOp (CatPrimOp $cst1, $cst2), $v)),
+  [(KnownWidth $v), (AnyConstantOp $cst1), (AnyConstantOp $cst2)]>;
```

Comment:
I think if `cst2` and `v` are signed, this will do the wrong thing, because the new outer cat will be UInt + SInt ?

Just curious: Why the constraint on knowing width of `v`?

Reply:
cat(sint, sint) -> uint.  Sigh.